### PR TITLE
🐛 ROSA: Fix missing permissions & Add external OIDC docs

### LINKED ISSUE #4931: ROSA: Document BreakGlassCredential for users
/kind documentation

https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4915 introduces support for external OIDC authentication, which includes a "break glass credential" used to configure RBAC within the cluster.

The credential's use and lifecycle should be documented for users.

### PR DESCRIPTION
**Pull request by muraee**
**What type of PR is this?**

/kind bug
/kind documentation

**What this PR does / why we need it**:
- Add docs for ROSA External OIDC feature
- Add missing required permissions to read/list configMaps.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4931

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [x] includes documentation
- [x] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by muraee** [comment]
cc @vincepri @nrb

File: docs/book/src/topics/rosa/external-auth.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  ....
+```
+
+Note: `oidcProviders` only accepts one entry at the moment.
+
+## Accessing the cluster
+
+### Setting up RBAC
+
+When `enableExternalAuthProviders` is set to `true`, ROSA provider will generate a temporarily admin kubeconfig secert in the same namespace named `<cluster-name>-bootstrap-kubeconfig`. This kubeonconfig can be used to access the cluster to setup RBAC for oidc users/groups.
```
Comment (typeid) [inline docs/book/src/topics/rosa/external-auth.md:53]:
```suggestion
When `enableExternalAuthProviders` is set to `true`, ROSA provider will generate a temporary admin kubeconfig secret in the same namespace named `<cluster-name>-bootstrap-kubeconfig`. This kubeconfig can be used to access the cluster to setup RBAC for OIDC users/groups.
```

File: docs/book/src/topics/rosa/external-auth.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+For example, bind the `cluster-admin` to an oidc group, to give admin permissions to all users part of that group:
```
Comment (typeid) [inline docs/book/src/topics/rosa/external-auth.md:55]:
```suggestion
The following example binds the `cluster-admin` role to an OIDC group, giving all users in that group admin permissions.
```

File: docs/book/src/topics/rosa/external-auth.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+Note: The generated bootstrap kubeconfig is only valid for 24h, and will not be usable afterwards. However, users can opt to manually delete the secret object to trigger the generation of a new one which will be valid for another 24h.
+
+### Login using the cli
+
+The [kubelogin kubectl plugin](https://github.com/int128/kubelogin/tree/master) can be used to login with OIDC credentials using the cli. 
+
+### Configuring OpenShift Console
+
+The OpenShift Console needs to be configured before it can be used to authenticate and login to the cluster. 
+1. Setup a new client in your OIDC provider with the following Redirect URL `<console-url>/auth/callback`. You can find the console URL in the status field of the `ROSAControlPlane` once the cluster is ready:
```
Comment (typeid) [inline docs/book/src/topics/rosa/external-auth.md:72]:
```suggestion
1. Setup a new client in your OIDC provider with the following Redirect URL: `<console-url>/auth/callback`. You can find the console URL in the status field of the `ROSAControlPlane` once the cluster is ready:
```

File: docs/book/src/topics/rosa/external-auth.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta2
+    kind: ROSAControlPlane
+    metadata:
+      name: "capi-rosa-quickstart-control-plane"
+    spec:
+      enableExternalAuthProviders: true
+      externalAuthProviders:
+      - name: my-oidc-provider
+        issuer:
+          issuerURL: https://login.microsoftonline.com/<tenant-id>/v2.0 # e.g. if using Micorsoft Entra ID
```
Comment (typeid) [inline docs/book/src/topics/rosa/external-auth.md:94]:
```suggestion
          issuerURL: https://login.microsoftonline.com/<tenant-id>/v2.0 # e.g. if using Microsoft Entra ID
```

**Review by typeid — CHANGES_REQUESTED** [review]
Thanks for adding this @muraee :) Suggested some fixes on typos.

**Comment by damdo** [comment]
@muraee could you update the title to the bug emoji and mention it in the title too.
I think the fact it fixes the permissions should take the precedence over the docs addition.
Thanks!

**Comment by muraee** [comment]
@typeid thanks for the suggestions, applied :)

**Comment by muraee** [comment]
@damdo makes sense, I updated the title.

File: controlplane/rosa/api/v1beta2/external_auth_types.go [hunk 5]
```diff
@@ -240,3 +238,12 @@ type TokenRequiredClaim struct {
 	// +required
 	RequiredValue string `json:"requiredValue"`
 }
+
+// LocalObjectReference references an object in the same namespace.
+type LocalObjectReference struct {
```
Comment (nrb) [inline controlplane/rosa/api/v1beta2/external_auth_types.go:243]: Why redefine this? To make it required?

Comment (muraee) [reply controlplane/rosa/api/v1beta2/external_auth_types.go:243]: right

**Review by damdo — COMMENTED** [review]
/lgtm
/assign @richardcase @dlipovetsky @Ankitasw

**Comment by nrb** [comment]
@damdo I don't think #4915 was backported to 2.4 - do you think it should be?

**Comment by damdo** [comment]
Good catch @nrb 
This will then go into v2.5.0 only.
I'll close the cherry-pick PR when it auto-opens

**Review by vincepri — APPROVED** [review]

**Pull request merged**